### PR TITLE
Improve integration with new issue filing interfaces

### PR DESCRIPTION
**Pull request by jalkire**
#### PR checklist

- [ ] Run through of all [test scenarios](https://github.com/Microsoft/accessibility-insights-windows/blob/master/docs/Scenarios.md) completed?
- [ ] Does this address an existing issue? If yes, Issue# - 
- [ ] Includes UI changes?
  - [ ] Run the production version of Accessibility Insights for Windows against a version with changes.
  - [ ] Attach any screenshots / GIF's that are applicable.

Note - After the PR has been created, certain checks will be kicked off. All of these checks must pass before a merge. 

#### Describe the change
- Handle case where connection config is null in ConnectionControl.
- Re-selected saved issue reporter
- Fix serialization issues with Guid

### CODE REVIEW

File: src/AccessibilityInsights.SharedUx/Controls/SettingsTabs/ConnectionControl.xaml.cs [hunk 1]
```diff
@@ -90,7 +90,12 @@ public void UpdateConfigFromSelections(ConfigurationModel configuration)
             {
                 configuration.SelectedIssueReporter = selectedIssueReporter.StableIdentifier;
                 string seralizedConfigs = configuration.IssueReporterSerializedConfigs;
-                Dictionary<Guid, string> configs = JsonConvert.DeserializeObject<Dictionary<Guid, string>>(seralizedConfigs);
+                Dictionary<Guid, string> configs = new Dictionary<Guid, string>();
+
+                if (seralizedConfigs != null)
```
Comment (DaveTryon) [inline src/AccessibilityInsights.SharedUx/Controls/SettingsTabs/ConnectionControl.xaml.cs]: >seralizedConfigs [](start = 20, length = 16)

Typo (missing the 'i' in "serial")

File: src/AccessibilityInsights.SharedUx/Settings/ConfigurationModel.cs [hunk 2]
```diff
@@ -126,8 +126,8 @@ public string AppVersion
 
         public Guid SelectedIssueReporter
         {
-            get => GetDataValue<Guid>(keyIssueReporting);
-            set => SetDataValue<Guid>(keyIssueReporting, value);
+            get => Guid.Parse(GetDataValue<string>(keyIssueReporting));
+            set => SetDataValue<string>(keyIssueReporting, value.ToString());
```
Comment (DaveTryon) [inline src/AccessibilityInsights.SharedUx/Settings/ConfigurationModel.cs]: >ToString() [](start = 65, length = 10)

Guid.ToString supports an optional formatting parameter (see [https://docs.microsoft.com/en-us/dotnet/api/system.guid.tostring?view=netframework-4.7.2](https://docs.microsoft.com/en-us/dotnet/api/system.guid.tostring?view=netframework-4.7.2)). Are we OK with the default format?

File: src/AccessibilityInsights.SharedUx/Settings/ConfigurationConstants.cs [hunk 3]
```diff
@@ -71,7 +71,7 @@ public partial class ConfigurationModel
         private const string keyTestReportPath = "TestReportPath";
         private const string keyTreeViewMode = "TreeViewMode";
         private const string keyVersion = "Version";
-        private const string keyIssueReporting= "IssueReportinGuid";
+        private const string keyIssueReporting= "IssueReportingGuid";
```
Comment (DaveTryon) [inline src/AccessibilityInsights.SharedUx/Settings/ConfigurationConstants.cs]: >IssueReportingGuid [](start = 49, length = 18)

There should be a unit test that ensures that this data is correct in the JSON, and that unit test should need to change as a result of this. I see no unit test change, so it seems that perhaps the unit test was omitted originally. Can we please add it now?

File: src/AccessibilityInsights.SharedUx/Settings/ConfigurationModel.cs (hunk 2 above)
Comment (jalkire) [reply src/AccessibilityInsights.SharedUx/Settings/ConfigurationModel.cs]: Guid.Parse should work regardless of format, so I think we're fine with the default:
>This method can convert strings in any of the five formats produced by the ToString
- https://docs.microsoft.com/en-us/dotnet/api/system.guid.parse?view=netframework-4.7.2

File: src/AccessibilityInsights.SharedUx/Settings/ConfigurationConstants.cs (hunk 3 above)
Comment (jalkire) [reply src/AccessibilityInsights.SharedUx/Settings/ConfigurationConstants.cs]: @flyingUnderTheRadar is working on the necessary unit test updates for this generally. The ConfigurationModelTests have already been broken in the feature branch--I would rather have them fixed/updated all at once than start the process in this PR.

File: src/AccessibilityInsights.SharedUx/Controls/SettingsTabs/ConnectionControl.xaml.cs (hunk 1 above)
Comment (jalkire) [reply src/AccessibilityInsights.SharedUx/Controls/SettingsTabs/ConnectionControl.xaml.cs]: 👓

**Pull request merged**